tools/copyright_checker.js

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env -S deno run --unstable --allow-read --allow-run
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { getSources, ROOT_PATH } from "./util.js";
+
+async function checkCopyright() {
+  const sourceFiles = await getSources(ROOT_PATH, [
+    // js and ts
+    "*.js",
+    "*.ts",
+    ":!:.github/mtime_cache/action.js",
+    ":!:cli/tests/testdata/**",
+    ":!:cli/bench/testdata/**",
+    ":!:cli/tsc/dts/**",
+    ":!:cli/tsc/*typescript.js",
+    ":!:cli/tsc/compiler.d.ts",
+    ":!:test_util/wpt/**",
+    ":!:tools/**", // these files are starts with `#!/usr/bin/env`
+    ":!:cli/tools/init/templates/**",
+
+    // rust
+    "*.rs",
+    ":!:ops/optimizer_tests/**",
+
+    // toml
+    "*Cargo.toml",
+  ]);
+
+  if (!sourceFiles.length) {
+    return;
+  }
+
+  let totalCount = 0;
+  for (const file of sourceFiles) {
+    const fileText = await Deno.readTextFile(file);
```

Comment:
Bit of a micro-optimization, but instead of reading the entire file, we could read just the first part of it:

```ts
const buffer = new Uint8Array(1024);
const textDecoder = new TextDecoder();

async function readFirstPartOfFile(filePath: string) {
  const file = await Deno.open(filePath, { read: true });
  try {
    const byteCount = await file.read(buffer);
    return textDecoder.decode(buffer.slice(0, byteCount ?? 0));
  } finally {
    file.close();
  }
}
```